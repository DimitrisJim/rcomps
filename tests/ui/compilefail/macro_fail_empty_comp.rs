// Check cases that compilefail.
//
// See https://rustc-dev-guide.rust-lang.org/tests/adding.html#error-annotations
// for annotating the snippets with the expected error.
// ignore-windows
extern crate rcomps;
use rcomps::comp;

/// Evaluate all combinations for empty comprehensions.
fn test_fail_empty_comp(){
    comp!();   //~ ERROR Empty expression.
    comp!{};   //~ ERROR Empty expression.
    comp![];   //~ ERROR Empty expression.
    comp!(()); //~ ERROR Unable to parse expression.
    comp![()]; //~ ERROR Unable to parse expression.
    comp!{()}; //~ ERROR Unable to parse expression.
}

fn main(){}