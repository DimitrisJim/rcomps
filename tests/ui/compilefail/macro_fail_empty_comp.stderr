error: Empty expression.
  --> $DIR/macro_fail_empty_comp.rs:11:5
   |
11 |     comp!();   //~ ERROR Empty expression.
   |     ^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Empty expression.
  --> $DIR/macro_fail_empty_comp.rs:12:5
   |
12 |     comp!{};   //~ ERROR Empty expression.
   |     ^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Empty expression.
  --> $DIR/macro_fail_empty_comp.rs:13:5
   |
13 |     comp![];   //~ ERROR Empty expression.
   |     ^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unable to parse expression.
  --> $DIR/macro_fail_empty_comp.rs:14:5
   |
14 |     comp!(()); //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unable to parse expression.
  --> $DIR/macro_fail_empty_comp.rs:15:5
   |
15 |     comp![()]; //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unable to parse expression.
  --> $DIR/macro_fail_empty_comp.rs:16:5
   |
16 |     comp!{()}; //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 6 previous errors

